### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by charmoniumQ**

### CODE REVIEW

**Comment by wsherman64** [comment]
The version of the most basic plugin in the online IlliXR documentation has a bug in that the start() method of a `threadloop` style plugin should not be overridden.

Here is a replacement that I'm now using, which includes a couple additional comments, plus some output to demonstrate that the plugin works:

// A minimal/no-op IlliXR plugin                                                 
                                                                                 
#include "common/phonebook.hpp"                                                  
#include "common/plugin.hpp"                                                     
#include "common/threadloop.hpp"                                                 
                                                                                 
using namespace ILLIXR;                                                          
                                                                                 
/// Inherit from `plugin` if you don't need the threadloop                       
/// Inherit from `threadloop` to provide a new thread to perform the task        
class basic_plugin : public threadloop {                                         
public:                                                                          
      basic_plugin(std::string name_, phonebook* pb_)                              
        : threadloop{name_, pb_}                                                 
        {                                                                        
        printf("Starting Plugin basic_plugin\n");                                
        RAC_ERRNO_MSG("basic_plugin in constructor");                            
    }                                                                            
    virtual ~basic_plugin() override {                                           
        printf("Ending Plugin basic_plugin\n");                                  
        RAC_ERRNO_MSG("basic_plugin in destructor");                             
    }                                                                            
                                                                                 
    // Do not override the start() method (for `threadloop` style plugins)       
                                                                                 
    // _p_one_iteration() is called from with the threadloop start() method      
    void _p_one_iteration() override {                                           
        printf("Iteration of basic_plugin\n");                                   
        RAC_ERRNO_MSG("basic_plugin at end of _p_one_iteration");                
    }                                                                            
};                                                                               
                                                                                 
// This line makes the plugin importable by Spindle                              
PLUGIN_MAIN(basic_plugin); ```

**Comment by charmoniumQ** [comment]
@wsherman64, I removed the RAC_*, since we are moving away from that. But otherwise, I've incorporated those changes.

File: docs/writing_your_plugin.md [hunk 1]
```diff
@@ -59,48 +59,75 @@ To add your own functionality via the plugin interface:
...
 
-    -   If you need custom concurrency (more complicated than a loop),
-            triggered concurrency (by events fired in other plugins),
-            or no concurrency then your plugin class should extend [`plugin`][13].
+    -   If you need custom concurrency (more complicated than a loop), triggered
+            concurrency (by events fired in other plugins), or no concurrency
+            then your plugin class should extend [`plugin`][13]. Your code goes
+            in the `start` method.
+
+    - If you want to scheduled data-driven work in either case, call
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:79]: schedule

File: docs/writing_your_plugin.md [hunk 2]
```diff
@@ -59,48 +59,75 @@ To add your own functionality via the plugin interface:
...
 
-        - They **must** be joined-or-disowned at-or-before
+    - They **must** be joined-or-disowned at-or-before
           `plugin::stop()`. This allows ILLIXR to shutdown cleanly.
 
 1.  Write a file called `plugin.cpp` with this body, replacing every instance of `plugin_name`:
 
     <!--- language: lang-cpp -->
 
+        /// A minimal/no-op IlliXR plugin
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:94]: ILLIXR

File: docs/writing_your_plugin.md [hunk 3]
```diff
@@ -59,48 +59,75 @@ To add your own functionality via the plugin interface:
...
+        class basic_plugin : public threadloop {
         public:
-            plugin_name(std::string name_, phonebook* pb_)
+            basic_plugin(std::string name_, phonebook* pb_)
                 : threadloop{name_, pb_}
-                { }
-            virtual void start() override { }
-            virtual ~plugin_name() override { }
+            {
+                printf("Starting Plugin basic_plugin\n");
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:111]: Use `cout`.

File: docs/writing_your_plugin.md [hunk 4]
```diff
@@ -59,48 +59,75 @@ To add your own functionality via the plugin interface:
...
+            }
+
+            virtual ~basic_plugin() override {
+                printf("Ending Plugin basic_plugin\n");
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:115]: Use `cout`.

File: docs/writing_your_plugin.md [hunk 5]
```diff
@@ -59,48 +59,75 @@ To add your own functionality via the plugin interface:
...
+            }
+
+            /// For `threadloop` style plugins, do not override the start() method unless you know what you're doing!
+
+            /// _p_one_iteration() is called in a thread created by threadloop::start()
+            void _p_one_iteration() override {
+                std::cout << "This goes to the log" << std::endl;
+                std::cout << "This goes to the console" << std::endl;
```
Comment (mhuzai) [inline docs/writing_your_plugin.md:123]: I don't understand the distinction between log and console here given that both prints are to `cout`.

Comment (charmoniumQ) [reply docs/writing_your_plugin.md]: How about now?

**Review by mhuzai — APPROVED** [review]

**Review by qinjunj — APPROVED** [review]

**Pull request merged**